test/02_gsn_acl_factory.js

```diff
@@ -0,0 +1,117 @@
+    readerAcc = await web3.eth.accounts.create("secret");
+  });
+
+  it('should initialize the GSN for ACLFactory by configuring its RelayHub and funding it', async () => {
+    let txHub = await factory.initialize(RELAY_HUB);
+    assert.equal(txHub.receipt.status, true);
+
+    let hubAddr = await factory.getHubAddr();
+    assert.equal(hubAddr, RELAY_HUB);
+
+    const factoryAddr = await factory.address;
+
+    // Required so we can fund individual ACLs from ACLFactory automatically
+    const faucetTx = await web3.eth.sendTransaction({
+      from: ROOT_ACCOUNT,
+      to: factory.address,
+      value: web3.utils.toWei(FACTORY_ACL_FAUCET_FUNDING_ETH, "ether")
+    });
+    assert.equal(faucetTx.status, true);
+
+    // Required to register the Recipient in RelayHub
+    const balance = await fundRecipient(web3, {
+      recipient: factoryAddr,
+      relayHubAddress: RELAY_HUB,
+      amount: web3.utils.toWei(FACTORY_FUNDING_ETH, "ether"),
+      from: ROOT_ACCOUNT
+    });
+
+    assert.equal(balance, web3.utils.toWei(FACTORY_FUNDING_ETH, "ether"));
+  });
+
+  it('should deploy an ACL from a user without any funds for FREE using ACLFactory', async () => {
+    const isFactoryDeployed = await isRelayHubDeployedForRecipient(web3, factory.address);
+    assert.equal(isFactoryDeployed, true);
+
+    const factoryBalance = await getRecipientFunds(web3, factory.address);
+    assert.equal(factoryBalance, web3.utils.toWei(FACTORY_FUNDING_ETH, "ether"));
+
+    gsnCtx = await fromConnection(
+      web3.eth.currentProvider.host, {
+        gsn: {
+          dev: false,
+          signKey: emptyAcc.privateKey
+        }
+    });
+
+    web3 = gsnCtx.lib;
+    const factoryGSN = await new gsnCtx.lib.eth.Contract(factory.abi, factory.address);
+
+    const createNewACLRes = await factoryGSN.methods.newACL(emptyAcc.address).send({
+      from: emptyAcc.address,
+      gasPrice: GAS_PRICE,
+      gasLimit: "7000000",
+    });
+    assert.equal(createNewACLRes.status, true);
+
+    const newACLAddr = createNewACLRes.events['NewACL'].returnValues['addr'];
+
+    const isACLGSNReady = await isRelayHubDeployedForRecipient(web3, newACLAddr);
+    assert.equal(isACLGSNReady, true);
+
+    newACLGSN = await new gsnCtx.lib.eth.Contract(ACL.abi, newACLAddr);
+    const aclOwner = await newACLGSN.methods.getOwner().call();
+
+    const isACLOwnerAdmin = await newACLGSN.methods.hasAdmin(emptyAcc.address).call();
+
+    assert.equal(aclOwner, emptyAcc.address);
+    assert.equal(isACLOwnerAdmin, true);
+  });
+
+  it('should be possible for ACL owner, to control permissions (grandRead, etc) without any funds, for FREE', async () => {
+    let hasReadAccess = await newACLGSN.methods.hasRead(readerAcc.address).call();
+    assert.equal(hasReadAccess, false);
+
+    const grantReadRes = await newACLGSN.methods.grantRead(readerAcc.address).send({
+      from: emptyAcc.address,
+      gasPrice: GAS_PRICE,
+      gasLimit: "7000000",
```

Comment:
[Minor] Calcula estimatedGas instead of hard coded values

Reply:
Tried right now, doesn't work properly. I am keeping the hardcoded value at least we know the upper limit. Later we can revisit by optimizing the compiler and so on.

Reply:
Okay, but it is awkard as I did use it in many places and with no issues. NVM is a minor thought